### PR TITLE
New profile: vcs/repository-files for github

### PR DESCRIPTION
**Pull request by jnv**
## Description

This introduces new `vcs/repository-files` profile for listing and accessing files in a VCS repository. The initial implementation focuses on GitHub using their [repository contents API](https://docs.github.com/en/rest/repos/contents#get-repository-content).

## Motivation and Context

Asked for by a user via chat support.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: grid/vcs/repository-files/maps/github.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        input: {
+          repository: 'octocat/Spoon-Knife',
+          reference: 'test-branch',
+        },
+      });
+      expect(result.isOk()).toBe(true);
+      expect(result).toMatchSnapshot();
+    });
+
+    describe('errors handling', () => {
```
Comment (Jakub-Vacek) [inline grid/vcs/repository-files/maps/github.test.ts:62]: Maybe add test case for undefined owner? 🤔

File: grid/vcs/repository-files/maps/github.suma [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      entries = Array.isArray(body) ? body : [body]
+      mappedEntries = entries.map((entry) => {
+        // NOTE: Submodules are exposed as files with size = 0
+        const type = entry.type === 'dir' ? 'directory' : entry.type
+        return {
+          name: entry.name,
+          path: entry.path,
+          type: type,
+          size: type === 'directory' ? undefined : entry.size,
+          rawUrl: entry.download_url,
```
Comment (Jakub-Vacek) [inline grid/vcs/repository-files/maps/github.suma:44]: I think profile states that rawUrl is empty for directories. Maybe check it similarly to type?

File: grid/vcs/repository-files/maps/github.suma [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        }
+      })
+      map result {
+        entries = mappedEntries
```
Comment (Jakub-Vacek) [inline grid/vcs/repository-files/maps/github.suma:48]: Looks like pagination is missing 🤔

**Review by Jakub-Vacek — COMMENTED** [review]
Looks good 👍 Just a few minor things and it is good to go 🙂

File: grid/vcs/repository-files/maps/github.suma (hunk 2 above)
Comment (jnv) [reply grid/vcs/repository-files/maps/github.suma]: You are right, they are sending it as `null`. I will rather test the value itself for emptiness.

File: grid/vcs/repository-files/maps/github.suma (hunk 3 above)
Comment (jnv) [reply grid/vcs/repository-files/maps/github.suma:48]: Intentionally so, GitHub's endpoint doesn't support pagination. I added it for GitLab and (potentially) GitHub's GraphQL API.

File: grid/vcs/repository-files/maps/github.test.ts (hunk 1 above)
Comment (jnv) [reply grid/vcs/repository-files/maps/github.test.ts:62]: Added!

**Review by Jakub-Vacek — APPROVED** [review]
Looks ok 👍

**Review by janhalama — APPROVED** [review]
LGTM :shipit:

**Pull request merged**